docs/syntax_and_semantics/macros/hooks.md

```diff
@@ -103,3 +103,45 @@ Foo.new.bar
 ```
 
 The `print_methods` macro will be run as soon as it is encountered - and will print an empty list as there are no methods defined at that point. Once the second declaration of `Foo` is compiled the `finished` macro will be run, which will print `[bar]`.
+
+Depending on the macro hook used, a hook can either be stacked or overridden. 
+
+## Stacking
+When stacked, a hook is executed multiple times in it's defined context for as many times as the hook is defined. Consider the following example:
+```crystal
+# Stack the top-level finished macro
+macro finished
+  {% puts "I will execute!" %}
+end
+
+macro finished
+  {% puts "I will also execute!" %}
+end
+```
+In the above example, both `finished` macros will execute. Stacking works for the following hooks: `inherited`, `included`, `extended`, `method_added`, `finished`
+
+## Overriding
+Overriding can only be used with the `method_missing` macro. This will only execute the last defined macro `method_missing` in a context. For example:
```

Comment:
**polish:** Instead of describing "Overriding" as a feature that's only available for `method_missing`, describe it as behaviour of that macro hook.
```suggestion
A definition of the `method_missing` macro hook overrides any previous definition of this hook in the same context. Only the last defined macro executes. For example:
```